crates/storage/db/src/abstraction/database.rs

```diff
@@ -4,7 +4,10 @@ use crate::{
     transaction::{DbTx, DbTxMut},
     DatabaseError,
 };
-use std::{fmt::Debug, sync::Arc};
+use std::{fmt::Debug, sync::Arc, time::Duration};
+
+/// Time limit that pruning can run before it's forced to yield the [`DatabaseProviderRW`] hook.
+pub const DEFAULT_TIME_OUT_DURATION_PRUNER: Duration = Duration::from_millis(100);
```

Comment:
I think a cleaner way would be to introduce a struct containing both `delete_limit` representing the max number of items to delete and `time_limit` representing the max allowed time to spend on deleting, making both of them `Option`. And then pass this struct as an argument to `prune_table_with_range`.

We definitely want to control the pruning time outside of the database abstraction.